FeatureExtractor.py

```diff
@@ -7,6 +7,21 @@ def __init__(self, image,memory):
     self.image = image
     self.memory = memory
 
+  def create_dense_descriptor(self):
+    if self.memory==False:
+        imgGray=cv.imread(self.image.get_path(),0)
```

Comment:
I would prefer you set a variables conditional and write the code once.

`image = self.image if self.memory else imgGray`

and then write the code once instead of twice.